controllers/verticalscalingpolicy_controller.go

```diff
@@ -197,60 +234,73 @@ func (r *VerticalScalingPolicyReconciler) Reconcile(ctxt context.Context, req ct
-						driftedTargets[igName] = true
-						continue
+			if scaleDownOnCpuUtilizationPolicy != nil {
+				cpuBelowScaleDownThreshold.LastHeartbeatTime = metav1.Now()
+
+				if currentCapacityUtilization < float64(scaleDownOnCpuUtilizationPolicy.Value) { // Check if scale down is a possibility
+					if cpuBelowScaleDownThreshold.Status == corev1.ConditionFalse {
+						cpuBelowScaleDownThreshold.LastTransitionTime = metav1.Now()
+						cpuBelowScaleDownThreshold.Status = corev1.ConditionTrue
+					}
+
+					if time.Since(cpuBelowScaleDownThreshold.LastTransitionTime.Time) > time.Duration(scaleDownOnCpuUtilizationPolicy.PeriodSeconds) {
+						// Check if eventual scale up is a possibility
+						if currentCapacityUtilization/2 < float64(scaleUpOnCpuUtilizationPolicy.Value) {
+							r.ManagerContext.ComputedTypes[namespacedIGName] = instanceTypeRange.InstanceTypes[currInstanceTypeIndex-1]
+							driftedTargets[namespacedIGName] = true
+							continue
+						}
 					}
+				} else if cpuBelowScaleDownThreshold.Status == corev1.ConditionTrue {
+					cpuBelowScaleDownThreshold.LastTransitionTime = metav1.Now()
+					cpuBelowScaleDownThreshold.Status = corev1.ConditionFalse
 				}
 			}
 
-			if scaleDownOnMemoryUtilization != nil {
-				if currentMemoryUtilization < float64(scaleDownOnMemoryUtilization.Value) { // Check if scale down is a possibility
-					if currentMemoryUtilization/2 < float64(scaleUpOnMemoryUtilization.Value) { // Check if eventual scale up is a possibility
-						// Add periodSeconds logic here
-						r.ManagerContext.ComputedTypes[igName] = instanceTypeRange.InstanceTypes[currInstanceTypeIndex-1]
-						driftedTargets[igName] = true
-						continue
+			if scaleDownOnMemoryUtilizationPolicy != nil {
+				memoryBelowScaleDownThreshold.LastHeartbeatTime = metav1.Now()
+
+				if currentMemoryUtilization < float64(scaleDownOnMemoryUtilizationPolicy.Value) { // Check if scale down is a possibility
+					if memoryBelowScaleDownThreshold.Status == corev1.ConditionFalse {
+						memoryBelowScaleDownThreshold.LastTransitionTime = metav1.Now()
+						memoryBelowScaleDownThreshold.Status = corev1.ConditionTrue
 					}
+
+					if currentMemoryUtilization/2 < float64(scaleUpOnMemoryUtilizationPolicy.Value) { // Check if eventual scale up is a possibility
+						if time.Since(memoryBelowScaleDownThreshold.LastTransitionTime.Time) > time.Duration(scaleDownOnMemoryUtilizationPolicy.PeriodSeconds) {
+							r.ManagerContext.ComputedTypes[namespacedIGName] = instanceTypeRange.InstanceTypes[currInstanceTypeIndex-1]
+							driftedTargets[namespacedIGName] = true
+							continue
+						}
+					}
+				} else if memoryBelowScaleDownThreshold.Status == corev1.ConditionTrue {
+					memoryBelowScaleDownThreshold.LastTransitionTime = metav1.Now()
+					memoryBelowScaleDownThreshold.Status = corev1.ConditionFalse
 				}
 			}
 		}
+
+		vsp.Status.TargetStatuses[namespacedIGName] = &v1alpha1.TargetStatus{
```

Comment:
This is outside of the nil check for `vsp.Status` so we should check here as well?